addons/binding/org.openhab.binding.tesla/src/main/java/org/openhab/binding/tesla/handler/TeslaHandler.java

```diff
@@ -137,35 +137,42 @@ public void initialize() {
 
         logger.trace("Initializing the Tesla handler for {}", getThing().getUID());
 
+        updateStatus(ThingStatus.OFFLINE);
```

Comment:
Well, if you are still initializing your connection for the first time, you better don't set any status. If you know that within a few seconds you can set it to ONLINE or OFFLINE, better wait for it.